### DISCUSSION TITLE
Bug: Dynamic import of components violates the order of styles

### DISCUSSION THREAD

**Discussion by ermakov95** [vercel/next.js#55423] [Help]
### Summary

Dynamic import of components violates the order of styles.

![image](https://github.com/vercel/next.js/assets/89069249/71cd73cf-4276-482f-92c1-38af3fc40f4a)

The order of occurrence:
1. For example, there is a Checkbox component that can include a className prop for overwriting styles. 
2. I use it on some page with overwriting styles (for example, not to display it on a mobile screen).
3. Also on the page I use a component that I import dynamically (for example, a modal window). This component also uses a Checkbox.
4. After dynamic import, the Checkbox styles from the modal window are recorded a second time and break the order.

I'll try to make an example later.

**Comment by vignesh-gupta** [comment]
@ermakov95  can you provide a code to reproduce this? or steps to reproduce (if generic)
